README.adoc

```diff
@@ -1,34 +1,125 @@
+:hazelcast: Hazelcast IMDG
+:framework: Spring Boot
 
-= Title
+= Getting Started with Hazelcast using Spring Boot
 
-// Content entered directly below the header but before the first section heading is called the preamble.
+This guide will help you to start using Hazelcast by building a Spring Boot application.
+
+include::{templates-url}/what-is-hz.adoc[]
 
 include::{templates-url}/link-to-repo.adoc[]
 
 == What You’ll Learn
 
-// Define the problem and tell reader what he/she will learn from this guide.
+include::{templates-url}/microservices/what-you-will-learn.adoc[]
 
 == Prerequisites
 
-// Define the prerequisites for the guide.
+- ~15 minutes
+include::{templates-url}/microservices/prerequisites.adoc[]
+
+== Using Hazelcast in the Spring Boot Application
+
+include::{templates-url}/microservices/application-structure.adoc[]
+
+If Hazelcast is on the classpath and a suitable configuration is found, Spring Boot auto-configures a HazelcastInstance that you can inject in your application. In the `pom.xml` file, you can see Hazelcast is added as a dependency:
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.hazelcast</groupId>
+    <artifactId>hazelcast-all</artifactId>
+    <version>${version.hazelcast}</version>
+</dependency>
+----
+
+Hazelcast configuration (`hazelcast.xml`) is placed under `src/main/resources/` folder. Now, we only need to auto-wire `HazelcastInstance` bean in `CommandController` and use it to access to Hazelcast data structures:
+
+[source, java]
+----
+package guides.hazelcast.springboot;
+
+import com.hazelcast.core.HazelcastInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.ConcurrentMap;
+
+@RestController
+public class CommandController {
+    @Autowired
+    HazelcastInstance hazelcastInstance;
+
+    private ConcurrentMap<String,String> retrieveMap() {
+        return hazelcastInstance.getMap("map");
+    }
+
+    @RequestMapping("/put")
+    public CommandResponse put(@RequestParam(value = "key") String key, @RequestParam(value = "value") String value) {
+        retrieveMap().put(key, value);
+        return new CommandResponse(value);
+    }
+
+    @RequestMapping("/get")
+    public CommandResponse get(@RequestParam(value = "key") String key) {
+        String value = retrieveMap().get(key);
+        return new CommandResponse(value);
+    }
+}
+----
+
+== Running the Sample Application
+
+Run the application using Maven on a console window:
+
+----
+mvn spring-boot:run -Dserver.port=8080
+----
+
+Then run the application again on another console window. But this time, note that you need to set a different value for the `server.port`.
+
+----
+mvn spring-boot:run -Dserver.port=8081
+----
```

Comment:
I would put this mvn command into a macro so that `Running the Sample Application` section can be completely reused by Quarkus and Micronaut.

Reply:
I thought application run methods might be different. If we keep the same, yes, we can do it as a macro.

Reply:
I belive getting started guides for spring,quarkus,micronaut and others will be the same but we can see while creating the others.